### PR TITLE
Exclude reduce_by_segment test case that crashes compiler

### PR DESCRIPTION
**Pull request by mmichel11**
This PR updates the test macro that excludes compilation of code that crashes the compiler during optimization in `reduce_by_segment.pass`.

### CODE REVIEW

**Review by danhoeflinger — APPROVED** [review]
LGTM, but lets see if it resolves the failures in the CI before merging.

**Comment by mmichel11** [comment]
Compilation of this test in CI is still crashing. I will find out why.

**Comment by mmichel11** [comment]
The TBB backend tests are now passing.

**Review by danhoeflinger — APPROVED** [review]
LGTM

**Pull request merged**
